src/Runner.Common/RunnerServer.cs

```diff
@@ -334,5 +337,14 @@ public Task<TaskAgent> UpdateAgentUpdateStateAsync(int agentPoolId, int agentId,
             CheckConnection(RunnerConnectionType.Generic);
             return _genericTaskAgentClient.UpdateAgentUpdateStateAsync(agentPoolId, agentId, currentState);
         }
+
+        //-----------------------------------------------------------------
+        // Runner Auth Url
+        //-----------------------------------------------------------------
+        public Task<string> GetRunnerAuthUrlAsync(int runnerPoolId, int runnerId)
+        {
+            CheckConnection(RunnerConnectionType.MessageQueue);
+            return _messageTaskAgentClient.GetAgentAuthUrlAsync(runnerPoolId, runnerId);
```

Comment:
There is a FF on server to control the return value from the endpoint. When FF is off, the endpoint always return empty. When FF is on, the endpoint will return authorization url from Token service and this will trigger runner migration.